src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int BOARD = 101;
+    private static final int RADIUS = 51;
+    private static final int SET_OF_FIGURES = 5;
+    private final Random rand = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        int index = rand.nextInt(SET_OF_FIGURES);
+        int value = rand.nextInt(BOARD);
+        int radius = rand.nextInt(RADIUS);
+        String color = colorSupplier.getRandomColor();
+
+        switch (index) {
+            case 0:
+                figure = new Circle(radius, color);
+                break;
+            case 1:
+                figure = new Square(color, value);
+                break;
+            case 2:
+                figure = new IsoscelesTrapezoid(color, value, value, value);
```

Comment:
Passing the same value for different parameters of `IsoscelesTrapezoid` might not always make sense as it represents a trapezoid with two sides being equal. Consider using different values for the base sides and height to reflect the actual properties of an isosceles trapezoid.